### PR TITLE
CockroachDB 22.2, ME/IE tests + fixes

### PR DESCRIPTION
**Pull request by pimeys**
Engine tests are now green. I splitted 22.1 and 22.2 tests so you can run them separately, if needed. The changes we needed to do for 22.2:

- For migrations, every connection runs two settings before anything else:
  - `SET enable_implicit_transaction_for_batch_statements=false`, which resolves the two issues below
  - `SET use_declarative_schema_changer=off`, which allows us for example add enum columns. The new schema changer has certain bugs, and we'll be opting out from it until they're fixed.
- Sequence max value default is different in 22.2.
- For float arrays, `0.0` is rounded as `0` on introspection. This seems to be a bug in CDB. We can live with that, just have separate tests for the different versions.

All bugs are reported to CDB.

As an added bonus, we'll be moving CDB 22.1 tests from Buildkite to GitHub Actions. They run in 27 minutes here vs. over an hour in Buildkite.

Closes: https://github.com/prisma/prisma/issues/16909
Closes: https://github.com/prisma/prisma/issues/17055
Part of: https://github.com/prisma/prisma/issues/16291

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -101,11 +101,17 @@ start-postgres15:
 dev-postgres15: start-postgres15
 	cp $(CONFIG_PATH)/postgres15 $(CONFIG_FILE)
 
+start-cockroach_22_2:
+	docker compose -f docker-compose.yml up -d --remove-orphans cockroach_22_2
+
+dev-cockroach_22_2: start-cockroach_22_2
+	cp $(CONFIG_PATH)/cockroach22.2 $(CONFIG_FILE)
```
Comment (janpio) [inline Makefile:108]: For consistency in naming:
```suggestion
	cp $(CONFIG_PATH)/cockroach_22_2 $(CONFIG_FILE)
```

File: Makefile [hunk 2]
```diff
@@ -101,11 +101,17 @@ start-postgres15:
...
+
 start-cockroach_22_1_0:
 	docker compose -f docker-compose.yml up -d --remove-orphans cockroach_22_1_0
 
 dev-cockroach_22_1_0: start-cockroach_22_1_0
-	cp $(CONFIG_PATH)/cockroach $(CONFIG_FILE)
+	cp $(CONFIG_PATH)/cockroach22.1 $(CONFIG_FILE)
```
Comment (janpio) [inline Makefile:114]: For consistency in naming with other existing providers:

```suggestion
	cp $(CONFIG_PATH)/cockroach_22_1_0 $(CONFIG_FILE)
```

File: migration-engine/connectors/sql-migration-connector/src/flavour/postgres/connection.rs [hunk 3]
```diff
@@ -21,11 +21,28 @@ impl Connection {
...
                 .await
-                .map_err(quaint_err(&url))?,
-        ))
+                .map_err(quaint_err(&url))?;
+
+            // Until at least version 22.2.5, enums are not type-sensitive without this.
+            quaint
+                .raw_cmd("SET use_declarative_schema_changer=off")
+                .await
+                .map_err(quaint_err(&url))?;
```
Comment (tomhoule) [inline migration-engine/connectors/sql-migration-connector/src/flavour/postgres/connection.rs:42]: Could we do this in one roundtrip?

Comment (pimeys) [reply migration-engine/connectors/sql-migration-connector/src/flavour/postgres/connection.rs:42]: yes we can

File: .buildkite/engineer [hunk 4]
```diff
@@ -13,7 +13,7 @@ fi
 if ! type "engineer" > /dev/null; then
     # Setup Prisma engine build & test tool (engineer).
     set -e
-    curl --fail -sSL "https://prisma-engineer.s3-eu-west-1.amazonaws.com/1.48.0/latest/$OS/engineer.gz" --output engineer.gz
+    curl --fail -sSL "https://prisma-engineer.s3-eu-west-1.amazonaws.com/cdb-versioning/latest/$OS/engineer.gz" --output engineer.gz
```
Comment (janpio) [inline .buildkite/engineer:16]: Note to fix this before merge

File: introspection-engine/introspection-engine-tests/tests/cockroachdb/mod.rs [hunk 5]
```diff
@@ -79,6 +79,31 @@ async fn identity_introspects_to_sequence_with_default_settings(api: TestApi) {
...
+    );
+    "#;
+
+    api.raw_cmd(sql).await;
+
+    let result = api.introspect_dml().await.unwrap();
+
+    let expected = expect![[r#"
+        model myTable {
+          id   Int    @id @default(sequence(maxValue: 2147483647))
```
Comment (janpio) [inline introspection-engine/introspection-engine-tests/tests/cockroachdb/mod.rs:99]: meh, did they intentionally introduce this new weird `maxValue`?

Comment (pimeys) [reply introspection-engine/introspection-engine-tests/tests/cockroachdb/mod.rs:99]: Yes.

File: migration-engine/connectors/sql-migration-connector/src/flavour/postgres/connection.rs (hunk 3 above)
Comment (pimeys) [reply migration-engine/connectors/sql-migration-connector/src/flavour/postgres/connection.rs:42]: [`6914798` (#3742)](https://github.com/prisma/prisma-engines/pull/3742/commits/6914798be13ab94736a0a6e851dd9bb1f5f379c0)

File: introspection-engine/introspection-engine-tests/tests/cockroachdb/mod.rs [hunk 6]
```diff
@@ -138,7 +163,7 @@ async fn scalar_list_defaults_work(api: &TestApi) -> TestResult {
             text_c_escape TEXT[] NOT NULL DEFAULT E'{ \'abc\', \'def\' }',
             colors COLOR[] NOT NULL DEFAULT '{ RED, GREEN }',
             int_defaults INT4[] NOT NULL DEFAULT '{ 9, 12999, -4, 0, 1249849 }',
-            float_defaults DOUBLE PRECISION[] NOT NULL DEFAULT '{ 0, 9.12, 3.14, 0.1242, 124949.124949 }',
+            float_defaults DOUBLE PRECISION[] NOT NULL DEFAULT '{ 0.0, 9.12, 3.14, 0.1242, 124949.124949 }',
```
[the remark this answers is not in the document]
Comment (pimeys) [reply introspection-engine/introspection-engine-tests/tests/cockroachdb/mod.rs:166]: Wanted to keep both sides similar, CDB considers 0 and 0.0 the same in these cases.

**Review by Weakky — APPROVED** [review]
Looks great to me 💯

File: .buildkite/engineer (hunk 4 above)
Comment (pimeys) [reply .buildkite/engineer]: [`ba1ebcc` (#3742)](https://github.com/prisma/prisma-engines/pull/3742/commits/ba1ebcc90a1749d5cf044b04c5a3329ccb5a31a2)

**Pull request merged**